ItemBaseCollectionUtils.cs

```diff
@@ -0,0 +1,286 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace FortressCraft.Community.Utilities
+{
+	/// <summary>
+	///		Extension Methods and Helpers for the ItemBase Class
+	/// </summary>
+	public static partial class ItemBaseUtil
+	{
+		/// <summary>
+		///		Gets the amount of items and cubes in any type of Enumerable ItemBase
+		/// </summary>
+		/// <param name="items">The list of items to get the total count from</param>
+		/// <returns>The amount of Items and Cubes</returns>
+		public static Int32 GetItemCount(this IEnumerable<ItemBase> items)
+		{
+			return items.Sum(itemBase => itemBase.GetAmount());
+		}
+
+		/// <summary>
+		///		Gets the amount of items in any type of Enumerable ItemBase
+		/// </summary>
+		/// <param name="items">The list of items to get the total count from</param>
+		/// <param name="itemId">The unique id of the item to count</param>
+		/// <returns>The amount of Items</returns>
+		public static Int32 GetItemCount(this IEnumerable<ItemBase> items, Int32 itemId)
+		{
+			return items.Where(item => item.mnItemID == itemId).Sum(item => item.GetAmount());
```

Comment:
item.mnItemID == -1 for all ItemCubeStack items. This doesn't work for none cube items.